### ISSUE TITLE
About the invert res-block module

### ISSUE THREAD

**Issue by hq-liu** [tonylins/pytorch-mobilenet-v2#3]
Hi, I have a question about the invert res-block module in your code. When I implemented this model, I feel confused about how to build the invert res-block module.
I found that in your code, you use this:
                    self.use_res_connect = self.stride == 1 and inp == oup
to ensure that input channels match with output channels. However, the input channels and output channels are always different. Thus, seems that there is no use for the this skip connection because (inp == oup) is always false. 
Hope you can reply this issue, thanks very match.

**Comment by FatherOfHam** [comment]
input channel and output channel are always the same.

**Comment by tonylins** [comment]
For each inverted residual sequence, input channel and output channel are the same except for the first layer.

**Comment by hq-liu** [comment]
I have already understood, thanks anyway

**Comment by foreverYoungGitHub** [comment]
But does this is same with the original paper?

The paper indicated that 'when input layer depth is 0 the underlying conv is the identity function.'

I'm just curious that whether the architecture should add an extra conv node for the first layer shortcut.

**Comment by foreverYoungGitHub** [comment]
The other thing is that, even though it is not indicated in the paper. But making the batch norm before the conv is possible to improve the accuracy. just like the resnet v2.

**Comment by harryhan618** [comment]
@tonylins I don't think input channel and output channel number is the same according to the paper Table 2.
![image](https://user-images.githubusercontent.com/10130410/38454571-4ef384d2-3a9c-11e8-84bb-0588208bee8c.png)
So as for the shortcut, I think there is a 1*1 conv layer that is used to adapt the input channel num to the output.
